lib/galaxy/managers/workflows.py

```diff
@@ -274,6 +275,11 @@ def build_invocations_query(self, trans, stored_workflow_id=None, history_id=Non
                 model.WorkflowInvocation.table.c.history_id == history_id
             )
 
+        if dataset_id is not None:
+            invocations_query = invocations_query.join(
+                model.WorkflowInvocationOutputDatasetAssociation
+            ).filter(model.WorkflowInvocationOutputDatasetAssociation.table.c.dataset_id == dataset_id)
```

Comment:
This only works for simple workflow outputs (so not for any random dataset produced by a workflow). It also doesn't work for collection. I think filtering by job is the correct way to do this.

Reply:
ok, which is currently not yet possible right? So I could update this change from `dataset_id` to `job_id`?

Reply:
Yes, I think that would be fine. There's a 1 to 1 relationship between job and invocation, so maybe this would be more appropriate in https://github.com/galaxyproject/galaxy/blob/61fcd517d69e1cbd981c3328ca0ed3fa60af6914/lib/galaxy/webapps/galaxy/api/workflows.py#L934 ?

Reply:
hmm, but then we need to make the invocation_id optional? Otherwise, one needs to know the invocation_id upfront..

However, the `invocation_id` and potentially the `workflow_id` could also be returned on the show `job` endpoint, then there is no need for a filter.